packages/phpstan-rules/packages/cognitive-complexity/tests/AstCognitiveComplexityAnalyzer/AstCognitiveComplexityAnalyzerTest.php

```diff
@@ -65,16 +62,22 @@ public function provideTokensAndExpectedCognitiveComplexity(): Iterator
     /**
      * @return ClassMethod|Function_
      */
-    private function parseFileToFistFunctionLike(string $fileContent): ?Node
+    private function parseFileToFirstFunctionLike(string $fileContent): Node
     {
         $parserFactory = new ParserFactory();
         $parser = $parserFactory->create(ParserFactory::ONLY_PHP7);
         $nodes = $parser->parse($fileContent);
 
         $nodeFinder = new NodeFinder();
-        return $nodeFinder->findFirst(
+        $firstFunctionlike =  $nodeFinder->findFirst(
             (array) $nodes,
             fn (Node $node): bool => $node instanceof ClassMethod || $node instanceof Function_
         );
+
+        if (!$firstFunctionlike instanceof ClassMethod && !$firstFunctionlike instanceof Function_) {
```

Comment:
assert the return-type early, so we don't need to handle a possible null-value at the call-site